lib/portlayer/vspc/handlers.go

```diff
@@ -0,0 +1,195 @@
+	} else if isVmotionPeer(b) {
+		log.Infof("vspc received VMOTION-PEER command")
+		h.HandleVmotionPeer(w, b)
+	} else if isVMUUID(b) {
+		log.Infof("vspc received VMUUID command")
+		h.HandleVMUUID(w, tc, b)
+	} else if isVmotionComplete(b) {
+		log.Infof("vspc received VMOTION-COMPLETE command")
+		h.HandleVmotionComplete(w, tc, b)
+	} else if isVmotionAbort(b) {
+		log.Infof("vspc received VMOTION-ABORT command")
+		h.HandleVmotionAbort(w, b)
+	}
+}
+
+// HandleVMUUID handles the telnet vm-uuid response
+func (h *Handler) HandleVMUUID(w io.Writer, tc *telnet.Conn, b []byte) {
+	vmuuid := strings.Replace(string(b[3:len(b)-1]), " ", "", -1)
+	log.Infof("vmuuid of the connected containerVM: %s", vmuuid)
+	// check if there exists another vm with the same vmuuid
+	vm, exists := h.vspc.getVM(vmuuid)
+	if !exists {
+		// create a new vm associated with this telnet connection
+		vm = NewVM(tc)
+		log.Infof("attempting to connect to the attach server")
+		remoteConn, err := net.Dial("tcp", fmt.Sprintf("%s:%d", h.vspc.attachSrvrAddr, h.vspc.attachSrvrPort))
+		if err != nil {
+			log.Errorf("cannot connect to the attach-server: %v", err)
+		}
+		vm.remoteConn = remoteConn
+		vm.vmUUID = vmuuid
+		h.vspc.addVM(vmuuid, vm)
+		// relay Reads from the remote system connection to the telnet connection associated with this vm
+		go h.vspc.relayReads(vm, remoteConn)
+	} else { //the vm existed before and was shut down or vmotioned
+		log.Infof("established second serial-port telnet connection with vm (vmuuid: %s)", vm.vmUUID)
+		vm.Lock()
+		defer vm.Unlock()
+		vm.prevContainerConn = vm.containerConn
+		vm.containerConn = tc
+	}
+}
+
+// HandleKnownSuboptions handles the known suboptions telnet command
+func (h *Handler) HandleKnownSuboptions(w io.Writer, b []byte) {
+	log.Infof("handling KNOWN-SUBOPTIONS")
+	var resp []byte
+	suboptions := b[3 : len(b)-1]
+	resp = append(resp, []byte{telnet.Iac, telnet.Sb, VmwareExt, KnownSuboptions2}...)
+	resp = append(resp, suboptions...)
+	resp = append(resp, telnet.Iac, telnet.Se)
+	log.Debugf("response to KNOWN-SUBOPTIONS: %v", resp)
+
+	if bytes.IndexByte(suboptions, GetVMVCUUID) != -1 && bytes.IndexByte(suboptions, VMVCUUID) != -1 {
+		resp = append(resp, getVMUUID()...)
+	}
+	w.Write(resp)
+}
+
+// HandleDoProxy handles the DO-PROXY telnet command
+func (h *Handler) HandleDoProxy(w io.Writer, b []byte) {
+	log.Infof("handling DO-PROXY")
+	var resp []byte
+	resp = append(resp, []byte{telnet.Iac, telnet.Sb, VmwareExt, WillProxy, telnet.Iac, telnet.Se}...)
+	log.Debugf("response to DO-PROXY: %v", resp)
+	w.Write(resp)
+}
+
+// HandleVmotionBegin handles the VMOTION-BEGIN telnet command
+func (h *Handler) HandleVmotionBegin(w io.Writer, tc *telnet.Conn, b []byte) {
+	if vm, exists := h.vspc.vmFromTelnetConn(tc); exists {
+		vm.Lock()
+		vm.inVmotion = true
+		vm.Unlock()
+		ch := make(chan struct{})
+		vm.vmotionStarted <- ch
+		<-ch
+	}
+	log.Infof("handling VMOTION-BEGIN")
+	seq := b[3 : len(b)-1]
```

Comment:
Each slicing, even though all of them started at 3, represents different type of information base on the telnet command in question. Coming up with a proper function name would be challenging:)